extensions/wikia/WikiaNewFiles/WikiaNewFilesSpecialController.class.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * @ingroup SpecialPage
+ */
+class WikiaNewFilesSpecialController extends WikiaSpecialPageController {
+	const DEFAULT_LIMIT = 48;
+	const PAGE_PARAM = 'page';
+
+	public function __construct() {
+		parent::__construct( 'Newimages', '', false );
+	}
+
+	public function index() {
+		$this->setHeaders();
+		$this->getContext()->getOutput()->setPageTitle( wfMessage( 'wikianewfiles-title' ) );
+		$this->wg->SupressPageSubtitle = true;
+
+		$request = $this->getRequest();
+
+		Wikia::addAssetsToOutput( 'upload_photos_dialog_js' );
+		Wikia::addAssetsToOutput( 'upload_photos_dialog_scss' );
+
+		$pageNumber = $request->getInt( self::PAGE_PARAM );
+		$pageNumber = max( $pageNumber, 1 );
+
+		// Fetch data from DB
+		$newFilesModel = new WikiaNewFilesModel();
+		$images = $newFilesModel->getImagesPage( self::DEFAULT_LIMIT, $pageNumber );
+		$imageCount = $newFilesModel->getImageCount();
+
+		// Pagination
+		$paginatorUrl = '?page=%s';
+		$paginator = Paginator::newFromArray( $imageCount, self::DEFAULT_LIMIT );
+		$paginator->setActivePage( $pageNumber - 1 );
+		$this->getOutput()->addHeadItem( 'Paginator', $paginator->getHeadItem( $paginatorUrl ) );
```

Comment:
Ugh, just noticed that `WikiaSpecialPageController` uses `__call` to "inherit" from `SpecialPage` if the method can't be found.  Should probably still use `getContext` as you did above since its a bit more clear.  Definitely hate the `__call` method...